src/OAuth/OAuth1/Service/Geocaching.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace OAuth\OAuth1\Service;
+
+use OAuth\OAuth1\Signature\SignatureInterface;
+use OAuth\OAuth1\Token\StdOAuth1Token;
+use OAuth\Common\Http\Exception\TokenResponseException;
+use OAuth\Common\Http\Uri\Uri;
+use OAuth\Common\Consumer\CredentialsInterface;
+use OAuth\Common\Http\Uri\UriInterface;
+use OAuth\Common\Storage\TokenStorageInterface;
+use OAuth\Common\Http\Client\ClientInterface;
+use OAuth\Common\Exception\Exception;
+
+class Geocaching extends AbstractService
+{
+    public function __construct(
+        CredentialsInterface $credentials,
+        ClientInterface $httpClient,
+        TokenStorageInterface $storage,
+        SignatureInterface $signature,
+        UriInterface $baseApiUri = null
+    ) {
+        parent::__construct($credentials, $httpClient, $storage, $signature, $baseApiUri);
+        if ($baseApiUri === null) {
+            $this->setBaseApiUri();
+        }
+        $this->setEndPoint();
+    }
+
+    public function getRequestTokenEndpoint()
+    {
+        return $this->getEndPoint();
+    }
+
+    public function getAuthorizationEndpoint()
+    {
+        return $this->getEndPoint();
+    }
+
+    public function getAccessTokenEndpoint()
+    {
+        return $this->getEndPoint();
+    }
+
+    public function getEndPoint()
+    {
+        return new Uri($this->endPoint);
+    }
+
+    public function setEndPoint($endPoint = 'live')
+    {
+        switch($endPoint) {
+            case 'staging':
```

Comment:
I don't understand why I need to create class constants. Users can set a different endPoint according to the platform (staging, live, live_mobile for example). 
I can create constants for the differents URI.